client/src/components/squids/SquidForm.jsx

```diff
@@ -0,0 +1,136 @@
+/* eslint-disable jsx-a11y/no-noninteractive-element-to-interactive-role */
+import React, { useState } from "react"
+
+import { useForm } from "react-hook-form"
+import { useQueryClient, useMutation } from "react-query"
+
+import "../../style/form.pcss"
+import { ApiClient } from "../../backend/ApiClient"
+
+export const SquidForm = () => {
+  const {
+    register,
+    reset,
+    handleSubmit,
+    formState: { errors },
+  } = useForm()
+  const [submitMessage, setSubmitMessage] = useState("")
+
+  const resetForm = () => {
+    reset({
+      name: "",
+      species: "",
+      specialPower: "",
+      experiencePoints: "",
+    })
+    setSubmitMessage("")
+  }
+
+  const addSquidMutation = useMutation((formPayload) => {
+    ApiClient.post("/squids", formPayload)
+  })
+
+  const queryClient = useQueryClient()
+
+  const postSquid = (data) => {
+    addSquidMutation.mutate(data, {
+      onSuccess: () => {
```

Comment:
this could be passed in to a function defined in the `hooks` directory as an option.